CalPatRec/src/AgnosticHelixFinder_module.cc

```diff
@@ -1388,11 +1402,41 @@ namespace mu2e {
 
     if (eDepAvg > _maxEDepAvg) return;
 
+    // compute direction of propagation and make save decision
+    HelixTool helTool(&hseed,_tracker);
+    float tzSlope = 0.0;
+    float tzSlopeErr = 0.0;
+    float tzSlopeChi2 = 0.0;
+    helTool.dirOfProp(tzSlope, tzSlopeErr, tzSlopeChi2);
+    HelixRecoDir helDir(tzSlope, tzSlopeErr, tzSlopeChi2);
+    hseed._recoDir = helDir;
+    hseed._propDir = helDir.computeDirection(_tzSlopeSigThresh);
+    if (!validHelixDirection(hseed._propDir)) return;
+
     // push back the helix seed to the helix seed collection
     HSColl.emplace_back(hseed);
 
   }
 
+  //-----------------------------------------------------------------------------
+  // check if the propagation direction is among the desired save directions
+  //-----------------------------------------------------------------------------
+  bool AgnosticHelixFinder::validHelixDirection(HelixRecoDir::PropDir direction) {
+
+    std::string hDir;
+
+    if (direction == HelixRecoDir::downstream) hDir = "downstream";
+    if (direction == HelixRecoDir::upstream) hDir = "upstream";
+    if (direction == HelixRecoDir::ambiguous) hDir = "ambiguous";
+
+    for (size_t i=0; i<_validHelixDirections.size(); i++) {
+      if (_validHelixDirections.at(i) == hDir) return true;
```

Comment:
Using strings for this comparison is inefficient. Why not just test the value (as an int) is within the enum range?
What situation is this logic trying to protect against? Normally, an enum should never have  a value outside its pre-defined range, unless you have hard-cast it from an int. If you can avoid that use case you can avoid even needing this function.